aws-predictions/src/androidTest/java/com/amplifyframework/predictions/aws/TestConfiguration.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.predictions.aws;
+
+import android.content.Context;
+import androidx.annotation.NonNull;
+import androidx.test.core.app.ApplicationProvider;
+
+import com.amplifyframework.AmplifyException;
+import com.amplifyframework.core.Amplify;
+import com.amplifyframework.core.AmplifyConfiguration;
+import com.amplifyframework.testutils.Resources;
+import com.amplifyframework.testutils.sync.SynchronousMobileClient;
+
+import com.amazonaws.mobile.config.AWSConfiguration;
+
+/**
+ * This is a class to help configure Amplify and prepare other test resources
+ * for Predictions integration tests.
+ */
+final class TestConfiguration {
+
+    private static final String AMPLIFY_CONFIGURATION_IDENTIFIER = "amplifyconfiguration";
+    private static final String AWS_CONFIGURATION_IDENTIFIER = "awsconfiguration";
+
+    private static TestConfiguration singleton;
+
+    private TestConfiguration() throws Exception {
+        Context context = ApplicationProvider.getApplicationContext();
+
+        Amplify.addPlugin(new AWSPredictionsPlugin());
+        configureAmplify(context);
+        setUpCredentials(context);
+    }
+
+    /**
+     * Process-wide configuration for the Storage instrumentation tests.
+     * @return A TestConfiguration instance
+     * @throws Exception if configuration fails
+     */
+    @NonNull
+    static synchronized TestConfiguration configureIfNotConfigured() throws Exception {
+        if (singleton == null) {
+            singleton = new TestConfiguration();
+        }
+        return singleton;
+    }
+
+    private static void configureAmplify(Context context) throws AmplifyException {
+        // Obtain Amplify Configuration
+        final int configId = Resources.getRawResourceId(context, AMPLIFY_CONFIGURATION_IDENTIFIER);
+        AmplifyConfiguration configuration = AmplifyConfiguration.fromConfigFile(context, configId);
```

Comment:
I have this here so that a customer can try running the instrumentation test with multiple versions of `amplifyconfiguration.json`. All they need to do to test using a different configuration would be to change the name of constant variable inside `TestConfiguration`. In fact, I may actually require this feature myself to test different types of interpret configuration since CLI officially supports one interpret resource per project.

Reply:
Ah, gotcha, hm, that makes sense.

To test multiple configs, we'll either have to build some teardown in to the `Amplify` facade, or not use it, in tests. (This way, we're not "stuck" with whatever the first instance of `new PredictionsCategory()` happens to have, in that `static` field inside of `Amplify`.) We could just rework our tests to go against the _`Category`_, directly. This would make our tests much more of real-life "integration" tests.

In other words, like:
```
PredictionsCategory predictions = new PredictionsCategory();
predictions.addPlugin(new AWSPredictionsPlugin());
CategoryConfiguration predictionsConfiguration =
    AmplifyConfiguration.fromConfigFile(context, Resources.getRawResourceId(".../path/..."))
        .forCategoryType(CategoryType.PREDICTIONS);
predictions.configure(predictionsConfiguration, context);
predictions.initialize(context);
```